### PR TITLE
Feature remove submodules and use spinouts

### LINKED ISSUE #54: Submodule
The submodule copy of shakenfist adds extra steps for first time users and annoys me during development.

I suggest that it be removed - thus removing two installation steps. In place, the playbook will git clone and checkout the requested tag/branch in the same directory (with a .gitignore directive for that directory). Also add an option for developers to specify a path to a local copy.

Yes nods?

### PR DESCRIPTION
**Pull request by mikalstill**
Hi. This patch is part of a series which spins out the ansible modules and python client into their own git repositories, rejigs the deployer to handle that change, adds richer git release descriptions, and fixes some minor bugs along the way. Please review the changes in the following order:

 - ansible-module spin out: https://github.com/shakenfist/ansible-modules/pull/1
 - client-python spin out: https://github.com/shakenfist/client-python/pull/1
 - deploy updates to use spin outs and support fancier git deploys: https://github.com/shakenfist/deploy/pull/58 (this PR)
 - removal of redundant code from shakenfist: https://github.com/shakenfist/shakenfist/pull/300

Fixes #54.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,34 @@
...
+### Common first steps
 
 ```bash
 sudo apt-get install ansible tox pwgen build-essential python3-dev python3-wheel
 git clone https://github.com/shakenfist/deploy
-cd deploy
-git submodule init
-git submodule update
-cd ansible
+cd deploy/ansible
```
Comment (mandoonandy) [inline README.md]: Is this step required for ansible-galaxy installation?

File: README.md [hunk 2]
```diff
@@ -37,50 +41,69 @@ gcloud compute images create sf-image \
...
-Virtual machines hosted under ESXi need two CPU options enabled.
+```bash
+$ cat ~/sf-deploys/cbr-remote.sh 
+#!/bin/bash
 
-```
-Hardware virtualization:
-    Expose hardware assisted virtualization to the guest OS
+export CLOUD=metal
+export VARIABLES="ADMIN_PASSWORD=p4ssw0rd FLOATING_IP_BLOCK=192.168.20.0/24"
```
Comment (mandoonandy) [inline README.md]: Setting ADMIN_PASSWORD in $VARIABLES leads to duplication relying on Ansibles's case-insensitvity and precedence. 
I'm thinking these two values should be set as individual environment variables.

**Review by mandoonandy — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mikalstill) [reply README.md]: No, it turns out it isn't. I'll remove it.

File: README.md (hunk 2 above)
Comment (mikalstill) [reply README.md]: So something more like:

```
export ADMIN_PASSWORD=p4ssw0rd
export FLOATING_IP_BLOCK="192.168.20.0/24"
```

?

Comment (mandoonandy) [reply README.md]: yep. that was my intention.

File: README.md (hunk 1 above)
Comment (mikalstill) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (mikalstill) [reply README.md]: Done.

**Review by mandoonandy — APPROVED** [review]
LGTM

**Pull request merged**